From 2bf3e2bbfae81ab50d141571414c0e6556bc0e0c Mon Sep 17 00:00:00 2001
From: Cheng-mean Liu <soccerl@microsoft.com>
Date: Thu, 11 Jan 2018 10:02:52 -0800
Subject: [PATCH] reduced the ND_MIN_NAMESPACE_SIZE from 4MB to 4KB

Signed-off-by: Cheng-mean Liu <soccerl@microsoft.com>
---
 ndctl/ndctl.h    | 2 +-
 test/dpa-alloc.c | 6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/ndctl/ndctl.h b/ndctl/ndctl.h
index 5e6905c..8c14d90 100644
--- a/ndctl/ndctl.h
+++ b/ndctl/ndctl.h
@@ -263,7 +263,7 @@ enum nd_driver_flags {
 };
 
 enum {
-	ND_MIN_NAMESPACE_SIZE = 0x00400000,
+	ND_MIN_NAMESPACE_SIZE = 0x00001000,
 };
 
 enum ars_masks {
diff --git a/test/dpa-alloc.c b/test/dpa-alloc.c
index d13cf5d..ba3deed 100644
--- a/test/dpa-alloc.c
+++ b/test/dpa-alloc.c
@@ -237,6 +237,12 @@ static int do_test(struct ndctl_ctx *ctx, struct ndctl_test *test)
 
 		uuid_unparse(namespaces[i].uuid, uuid_str);
 		size = ndctl_namespace_get_size(victim);
+
+		rc = ndctl_namespace_disable_invalidate(victim);
+                if (rc) {
+			fprintf(stderr, "failed to disable %s\n", uuid_str);
+			return rc;
+		}
 		rc = ndctl_namespace_delete(victim);
 		if (rc) {
 			fprintf(stderr, "failed to delete %s\n", uuid_str);
-- 
2.11.0

